cmd/otel-allocator/server/server.go

```diff
@@ -172,15 +208,38 @@ func (s *Server) UpdateScrapeConfigResponse(configs map[string]*promconfig.Scrap
 	}
 
 	s.mtx.Lock()
-	s.scrapeConfigResponse = jsonConfigNew
+	if marshalSecretValue {
+		s.ScrapeConfigMarshalledSecretResponse = jsonConfigNew
+	} else {
+		s.scrapeConfigResponse = jsonConfigNew
+	}
 	s.mtx.Unlock()
+
+	return nil
+}
+
+// UpdateScrapeConfigResponse updates the scrape config response. The target allocator first marshals these
+// configurations such that the underlying prometheus marshaling is used. After that, the YAML is converted
+// in to a JSON format for consumers to use.
+func (s *Server) UpdateScrapeConfigResponse(configs map[string]*promconfig.ScrapeConfig) error {
+	err := s.MarshalScrapeConfig(configs, false)
+	if err != nil {
+		return err
+	}
+	err = s.MarshalScrapeConfig(configs, true)
+	if err != nil {
+		return err
+	}
```

Comment:
should we switch on if we're on a secure server here instead of trying one and then the other?

Reply:
As we are having both secure and insecure servers running simultaneously, it made sense to keep (and update) both versions

Reply:
wouldn't this result in us dual writing the scrape config though? Is there no way for us to know if it's the https or http server responding?

Reply:
It will. We are writing both but serving the relevant version based on if the connection is secure or not - https://github.com/open-telemetry/opentelemetry-operator/pull/2921/files/70929155ab0a15976a3180f6524e5c5d5db70128#diff-d586dd0d67f5b7b16ee4093cad5611ced5099b2e6c31e2a47f07248c6048ddaaR240

Reply:
ohhhh i see now, thanks for explaining :)